### PR TITLE
Add crash header to 500 responses

### PR DESCRIPTION
**Pull request by asriniva**
Adds the equivalent of this: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/master/src/logger.ts#L25

### CODE REVIEW

File: src/functions_framework/__init__.py [hunk 1]
```diff
@@ -160,6 +163,13 @@ def read_request(response):
     return response
 
 
+def crash_handler(e):
+    """
+    Return crash header to allow logging 'crash' message in logs.
+    """
+    return str(e), 500, {_FUNCTION_STATUS_HEADER_FIELD: _CRASH}
```
Comment (di) [inline src/functions_framework/__init__.py:170]: Should this be all 500-level status codes or just `500`?

File: src/functions_framework/__init__.py [hunk 2]
```diff
@@ -215,6 +225,7 @@ def create_app(target=None, source=None, signature_type=None):
     # 5. Create the application
     app = flask.Flask(target, template_folder=template_folder)
     app.config["MAX_CONTENT_LENGTH"] = MAX_CONTENT_LENGTH
+    app.register_error_handler(500, crash_handler)
```
Comment (di) [inline src/functions_framework/__init__.py:228]: Should this be all 500-level status codes or just `500`?

Comment (asriniva) [reply src/functions_framework/__init__.py:228]: This is primarily for uncaught exceptions, which produce a 'crash' message in GCF Python 3.7. From my understanding, all of these produce 500 response codes in Flask?

Comment (di) [reply src/functions_framework/__init__.py:228]: Yep

**Review by di — APPROVED** [review]

**Pull request merged**